components/tikv_util/src/lib.rs

```diff
@@ -462,6 +462,22 @@ pub fn set_panic_hook(panic_abort: bool, data_dir: &str) {
     let orig_hook = panic::take_hook();
     panic::set_hook(Box::new(move |info: &panic::PanicInfo<'_>| {
         use slog::Drain;
+
+        // There might be remaining logs in the async logger.
```

Comment:
How does it work by only moving those codes?

Reply:
The panic info will be logged here: https://github.com/tikv/tikv/blob/58ca498ff7bb59975b517b4e02d495cff8c74de0/components/tikv_util/src/lib.rs#L479 

What the code I moved do is to switch the global logger to a terminal logger, which wrappers `stderr` as a drainer. The `term_drainer` is here: https://github.com/tikv/tikv/blob/58ca498ff7bb59975b517b4e02d495cff8c74de0/components/tikv_util/src/logger/mod.rs#L139

And @hicqu have a comment here: https://github.com/tikv/tikv/pull/7336#discussion_r404717401
so log the panic info to stderr should be sufficient.